R/generateIIASASubmission.R

```diff
@@ -105,7 +107,11 @@ generateIIASASubmission <- function(mifs = ".", # nolint cyclocomp_linter
         "piam_variable" = removePlus(.data$piam_variable),
         "piam_factor" = ifelse(is.na(.data$piam_factor), 1, as.numeric(.data$piam_factor))
       ) %>%
-      select("variable", "unit", "piam_variable", "piam_unit", "piam_factor")
+      dplyr::bind_rows(tibble("weight" = "NULL")) %>% # add the optional weight column if not present
```

Comment:
My solution for making the weight column optional - is there a more elegant way to achieve this?

Reply:
Well, if it works, it seems ok to me

Reply:
So far, we have been pretty rigid with optional columns. They don't exist. This solution would require us to add the weight column in all mappings and make it mandatory. Has the advantage that we don't have to keep in mind that it may or may not be around.

@orichters any objections against introducing the weight column in all templates?

Reply:
Maybe I understood the solution by @jdrtommey differently, namely that he adds the column here if not present. So we would not need to add it to all mappings, which is really what I prefer.

Reply:
That's what is happening here. If it is your preferred solution, then it is fine by me.